### PR TITLE
Frontend: Sets up MetricConfig for Maternal Mortality

### LINKED ISSUE #3306: Setup MetricConfig for MaternalHealth
## Setting Up the Metric Config

- [x]  For each new data type, add an entry to an existing MetricConfig file, or create a new `[Category]MetricConfig.ts` file and place them all in there. Each array will translate into a dropdown option of the first object; and placing more than one MetricConfig object in the array will make those configs data type sub-dropdown options (e.g. Jail and Prison, both options under Incarceration).
- [x]  Import the specific metric configuration for your new dataset into the main metric configuration file (e.g., `MetricConfig.ts`).
- [x]  Within your dataset's configuration file, define unique identifiers (IDs) for each metric and their types (e.g., count, percentage, rate).
- [x]  For each metric, specify how it should be displayed. This includes setting titles for charts, labels for data points, and any special formatting needs.
- [x] Topic definition: "Maternal deaths per 100,000 live births. Births were modeled using counts of live births among individuals aged 10 to 54 years between 1999 and 2019. Deaths were modeled from death certificate data for pregnant or recently pregnant individuals aged 10 to 54 years. Maternal deaths are coded as deaths that occurred up to 1 year after the end of pregnancy and were coded with the use of the US standard pregnancy question and/or a specific International Statistical Classification of Diseases and Related Health Problems codes."
- [x]  Configure how it should be mapped, including color schemes and value ranges (if applicable).
- [x]  Ensure that the frontend components (e.g., charts, tables, maps) are aware of and can pull configuration from this setup.

### PR DESCRIPTION
**Pull request by JDemlow**
# Description and Motivation

Closes #3306 

## Has this been tested? How?

Using integrated terminal and browser

## Screenshots (if appropriate)

Screenshot (133)

## Types of changes

(leave all that apply)

- New content or feature

### CODE REVIEW

File: frontend/src/data/providers/GunViolenceYouthProvider.ts
Comment (benhammondmusic) [inline frontend/src/data/providers/GunViolenceYouthProvider.ts:1]: Do you mind removing these gun violence files from this PR? I think it's mostly linting stuff but definitely best practice to roll these changes back and keep the PR totally Focused on maternal mortality frontend stuff. 

A quick way to roll back is locally checkout the file from origin/main like `git checkout origin/main -- some/weird/unrelated/file.tsx`

**Review by benhammondmusic — CHANGES_REQUESTED** [review]

File: frontend/src/data/providers/GunViolenceYouthProvider.ts
Comment (JDemlow) [reply frontend/src/data/providers/GunViolenceYouthProvider.ts:1]: Sorry, did you want all of the Gun Violence files removed from the provider folder, or just the specified GunViolenceYouthProvider.ts?

**Comment by JDemlow** [comment]
Only files showing as changed are related to MM. Should be good now!

File: frontend/src/data/providers/MaternalMortalityProvider.ts [hunk 1]
```diff
@@ -1,69 +1,105 @@
...
-        let df = maternalMortalityDataset.toDataFrame();
+  async getDataInternal(metricQuery: MetricQuery): Promise<MetricQueryResponse> {
+  try {
+    const breakdowns = metricQuery.breakdowns
+    const datasetId = this.getDatasetId(breakdowns, undefined, metricQuery.timeView)
+    
+    if (!datasetId) throw Error('DatasetId is undefined')
+    
+    const specificDatasetId = appendFipsIfNeeded(datasetId, breakdowns)
+    console.log('Loading dataset with ID:', specificDatasetId) // Debugging log
```
Comment (benhammondmusic) [inline frontend/src/data/providers/MaternalMortalityProvider.ts:62]: let's make sure we remove all these logs

File: frontend/src/data/providers/MaternalMortalityProvider.ts [hunk 2]
```diff
@@ -1,69 +1,105 @@
...
-        }
-        df = this.renameGeoColumns(df, breakdowns);
+    console.log('Data loaded:', df.toArray()) // Check if data is loaded
 
-        // Apply demographic breakdown filters
-        df = this.applyDemographicBreakdownFilters(df, breakdowns);
-        df = this.removeUnrequestedColumns(df, metricQuery);
+    // Filter by geography
+    df = this.filterByGeo(df, breakdowns)
+    console.log('After filtering by geo:', df.toArray()) // Check data after filtering by geography
```
Comment (benhammondmusic) [inline frontend/src/data/providers/MaternalMortalityProvider.ts:72]: remove

File: frontend/src/data/providers/MaternalMortalityProvider.ts [hunk 3]
```diff
@@ -1,69 +1,105 @@
...
 
-        return new MetricQueryResponse(df.toArray(), consumedDatasetIds);
+    if (df.toArray().length === 0) {
+      console.warn('No data after filtering, returning empty response')
```
Comment (benhammondmusic) [inline frontend/src/data/providers/MaternalMortalityProvider.ts:75]: remove

File: frontend/src/data/providers/MaternalMortalityProvider.ts [hunk 4]
```diff
@@ -1,69 +1,105 @@
...
+      return new MetricQueryResponse([], consumedDatasetIds)
     }
 
-    allowsBreakdowns(breakdowns: Breakdowns): boolean {
-        return breakdowns.hasExactlyOneDemographic();
-    }
+    df = this.renameGeoColumns(df, breakdowns)
+    df = this.applyDemographicBreakdownFilters(df, breakdowns)
+    console.log('After applying demographic breakdown filters:', df.toArray()) // Check data after applying demographic filters
```
Comment (benhammondmusic) [inline frontend/src/data/providers/MaternalMortalityProvider.ts:81]: remove

File: frontend/src/data/providers/MaternalMortalityProvider.ts [hunk 5]
```diff
@@ -1,69 +1,105 @@
...
+    
+    df = this.removeUnrequestedColumns(df, metricQuery)
+    console.log('Final data returned:', df.toArray()) // Final check before returning data
```
Comment (benhammondmusic) [inline frontend/src/data/providers/MaternalMortalityProvider.ts:84]: remove

**Review by benhammondmusic — APPROVED** [review]
great! just remove those console logs/warns and then merge it in! thanks :)

**Pull request merged**